src/app/subnets/views/VLANDetails/VLANActionForms/VLANActionForms.tsx

```diff
@@ -0,0 +1,28 @@
+import type { SetSidePanelContent } from "@/app/base/side-panel-context";
+import ReservedRangeDeleteForm from "@/app/subnets/components/ReservedRangeDeleteForm";
+import ReservedRangeForm from "@/app/subnets/components/ReservedRangeForm";
+import VLANDeleteForm from "@/app/subnets/views/VLANDetails/VLANDeleteForm";
+import { VLANActionTypes } from "@/app/subnets/views/VLANDetails/constants";
+import type { VLANActionType } from "@/app/subnets/views/VLANDetails/constants";
+
+type VLANActionFormProps = {
+  id: number;
+  activeForm: VLANActionType;
+  setSidePanelContent: SetSidePanelContent;
+};
+
+const VLANActionForms = ({
+  id,
+  activeForm,
+  setSidePanelContent,
+}: VLANActionFormProps) => {
+  return activeForm === VLANActionTypes.DeleteVLAN ? (
+    <VLANDeleteForm closeForm={() => setSidePanelContent(null)} id={id} />
+  ) : activeForm === VLANActionTypes.ReserveRange ? (
+    <ReservedRangeForm setActiveForm={setSidePanelContent} />
+  ) : activeForm === VLANActionTypes.DeleteReservedRange ? (
+    <ReservedRangeDeleteForm setActiveForm={setSidePanelContent} />
+  ) : null;
```

Comment:
We should probably represent this better, it's difficult to read

Reply:
You're right, wanted to limit the scope as it's a bugfix. Updated.

Reply:
Agreed, a simple switch/case would be much more readable here